spec/factories/assignments.rb

```diff
@@ -17,12 +17,12 @@
 
     after(:build) do |assignment, evaluator|
       if evaluator.assignment_properties_attributes
-        assignment.assignment_properties ||= build(:assignment_properties,
+        assignment.assignment_properties = build(:assignment_properties,
                                                    assignment: assignment,
```

Comment:
You should fix the indentation in this line and the lines below (and note the same is true with the other `build` call).